src/gc/gc.d

```diff
@@ -605,61 +605,73 @@ struct GC
             else
             {
                 auto pool = gcx.findPool(p);
-                psize = pool.getSize(p);     // get allocated size
-                if (psize >= PAGESIZE && size >= PAGESIZE)
+                if (pool.isLargeObject)
                 {
+                    if (size <= PAGESIZE / 2)
+                        goto Lmalloc; // switching from large object pool to small object pool
```

Comment:
This is an actual change, but I think the original version is inaccurate: pages from the large object pools are used for allocation larger than 2048.